### PR TITLE
Update mpas-source: coastal merge

### PR DESCRIPTION
**Pull request by mark-petersen**
Additions for the coastal needs. Includes three new analysis members:

- harmonic_analysis
- sediment_flux_index
- sediment_transport

and changes to the hurricane and drying slope COMPASS test cases and init mode. Does not change forward mode simulations.

[BFB]
[NML]

### CODE REVIEW

**Comment by jonbob** [comment]
@mark-petersen - is there a need for the new analysis members to be included in E3SM at this point? They will also need to be added to the streams file generation, if so.

**Comment by mark-petersen** [comment]
Rebased. @jonbob I'm seeing an build error. I'm looking into it now, and will post when I'm done.

**Comment by jonbob** [comment]
@mark-petersen - there is a new namelist group for vegetation_drag. Do we need to include it in E3SM? We have been removing all references for groups time_varying_forcing, tidal_forcing, tidal_potential_forcing, and wetting_drying -- are any of those needed for this PR?

**Comment by mark-petersen** [comment]
I was planning to have the lists below use the MPAS-O defaults and be off by default, but actually appear in the E3SM mpaso_in namelist. @sbrus89, is that what you were thinking too? 

<details>
<summary> namelist additions </summary> <p>

```
&vegetation_drag
    config_use_vegetation_drag = .false.
    config_use_vegetation_manning_equation = .false.
    config_vegetation_drag_coefficient = 1.09
/
...
&AM_sedimentFluxIndex
    config_AM_sedimentFluxIndex_enable = .false.
    config_AM_sedimentFluxIndex_compute_on_startup = .true.
    config_AM_sedimentFluxIndex_write_on_startup = .true.
    config_AM_sedimentFluxIndex_compute_interval = 'output_interval'
    config_AM_sedimentFluxIndex_output_stream = 'sedimentFluxIndexOutput'
    config_AM_sedimentFluxIndex_directory = 'analysis_members'
    config_AM_sedimentFluxIndex_use_lat_lon_coords = .true.
/
&AM_sedimentTransport
    config_AM_sedimentTransport_enable = .false.
    config_AM_sedimentTransport_compute_on_startup = .true.
    config_AM_sedimentTransport_write_on_startup = .true.
    config_AM_sedimentTransport_compute_interval = 'output_interval'
    config_AM_sedimentTransport_output_stream = 'sedimentTransportOutput'
    config_AM_sedimentTransport_directory = 'analysis_members'
    config_AM_sedimentTransport_grain_size = 2.5e-4
    config_AM_sedimentTransport_ws_formula = 'VanRijn1993'
    config_AM_sedimentTransport_bedld_formula = 'Soulsby-Damgaard'
    config_AM_sedimentTransport_SSC_ref_formula = 'Lee2004'
    config_AM_sedimentTransport_drag_coefficient = 2.5e-3
    config_AM_sedimentTransport_erate = 5.0e-4
    config_AM_sedimentTransport_tau_ce = 0.1
    config_AM_sedimentTransport_tau_cd = 0.1
    config_AM_sedimentTransport_Manning_coef = 0.022
    config_AM_sedimentTransport_grain_porosity = 0.5
    config_AM_sedimentTransport_water_density = 1020
    config_AM_sedimentTransport_grain_density = 2650
    config_AM_sedimentTransport_alpha = 1e-2
    config_AM_sedimentTransport_kinematic_viscosity = 1e-6
    config_AM_sedimentTransport_vertical_diffusion_coefficient = 1e-2
    config_AM_sedimentTransport_bedload = .true.
    config_AM_sedimentTransport_suspended = .true.
    config_AM_sedimentTransport_use_lat_lon_coords = .true.
/
&AM_harmonicAnalysis
    config_AM_harmonicAnalysis_enable = .false.
    config_AM_harmonicAnalysis_compute_interval = 'output_interval'
    config_AM_harmonicAnalysis_start = '0001-01-01_00:00:00'
    config_AM_harmonicAnalysis_end = '0001-01-01_00:00:00'
    config_AM_harmonicAnalysis_output_stream = 'harmonicAnalysisOutput'
    config_AM_harmonicAnalysis_restart_stream = 'harmonicAnalysisRestart'
    config_AM_harmonicAnalysis_compute_on_startup = .false.
    config_AM_harmonicAnalysis_write_on_startup = .false.
    config_AM_harmonicAnalysis_use_M2 = .true.
    config_AM_harmonicAnalysis_use_S2 = .true.
    config_AM_harmonicAnalysis_use_N2 = .true.
    config_AM_harmonicAnalysis_use_K2 = .true.
    config_AM_harmonicAnalysis_use_K1 = .true.
    config_AM_harmonicAnalysis_use_O1 = .true.
    config_AM_harmonicAnalysis_use_Q1 = .true.
    config_AM_harmonicAnalysis_use_P1 = .true.
```
</details>
I'm guessing we decided to remove all those other groups because they shouldn't be run in E3SM yet. Maybe we should review that decision. But for now, I think the four namelists above should be included.

I see there is a flag `config_AM_harmonicAnalysis_restart_stream = 'harmonicAnalysisRestart'`. I need to add that to the streams quick too.

**Comment by mark-petersen** [comment]
@sbrus89 could you check the streams additions in this PR? The variables are straight from your defaults, but good to double check.

**Comment by mark-petersen** [comment]
Tested the head, 2f2a938 that includes the stream file additions. Passes:
```
SMS_D_Ln9.T62_oQU120_ais20.MPAS_LISIO_TEST.cori-haswell_intel
SMS_D_Ln9.T62_oQU120_ais20.MPAS_LISIO_TEST.cori-haswell_gnu
SMS_D_Ln9.T62_oQU120_ais20.MPAS_LISIO_TEST.cori-knl_gnu
PET_Ln9.T62_oEC60to30v3wLI.GMPAS-DIB-IAF-ISMF.cori-knl_intel
```
That is without namelist additions, so it just using the defaults, so those new namelist groups are default off.

**Comment by sbrus89** [comment]
> @mark-petersen - there is a new namelist group for vegetation_drag. Do we need to include it in E3SM? We have been removing all references for groups time_varying_forcing, tidal_forcing, tidal_potential_forcing, and wetting_drying -- are any of those needed for this PR?

@jonbob, let's go ahead and use this list for removing namelist groups:
time_varying_forcing, tidal_forcing, vegetation_drag, and wetting_drying
(i.e. stop removing tidal_potential_forcing and start removing vegetation_drag)

File: components/mpas-ocean/cime_config/buildnml [hunk 1]
```diff
@@ -780,6 +780,101 @@ def buildnml(case, caseroot, compname):
...
+        lines.append('        type="input;output"')
+        lines.append('        io_type="{}"'.format(ocn_pio_typename))
+        lines.append('        filename_template="mpaso.rst.am.harmonicAnalysis.$Y-$M-$D_$S.nc"')
+        lines.append('        filename_interval="output_interval"')
+        lines.append('        clobber_mode="truncate"')
+        lines.append('        input_interval="none"')
+        lines.append('        packages="harmonicAnalysisAMPKG"')
+        lines.append('        reference_time="01-01-01_00:00:00"')
+        lines.append('        output_interval="none">')
+        lines.append('')
```
Comment (sbrus89) [inline components/mpas-ocean/cime_config/buildnml:873]: add `xtime` here

File: components/mpas-ocean/cime_config/buildnml [hunk 2]
```diff
@@ -780,6 +780,101 @@ def buildnml(case, caseroot, compname):
...
+        lines.append('        precision="single"')
+        lines.append('        io_type="{}"'.format(ocn_pio_typename))
+        lines.append('        filename_template="mpaso.hist.am.harmonicAnalysis.$Y-$M-$D.nc"')
+        lines.append('        filename_interval="00-01-00_00:00:00"')
+        lines.append('        reference_time="01-01-01_00:00:00"')
+        lines.append('        output_interval="none"')
+        lines.append('        clobber_mode="truncate"')
+        lines.append('        packages="harmonicAnalysisAMPKG">')
+        lines.append('')
+        lines.append('    <var name="xtime"/>')
```
Comment (sbrus89) [inline components/mpas-ocean/cime_config/buildnml:840]: leave `xtime` in

**Comment by mark-petersen** [comment]
@jonbob, @sbrus89 and I looked this over together, and it is ready now after the namelist scripts. I'm running a few tests with the most recent commit to be sure.

**Comment by jonbob** [comment]
@mark-petersen, @sbrus89 - the scripts updates to match the Registry changes are merged, if you want to look at them. Please be aware that any new variables will have the Registry default values unless we consciously change them in E3SM.

**Comment by jonbob** [comment]
At this point I'll merge this PR and test before pushing to next

**Comment by jonbob** [comment]
test merge passes:
* SMS_D.ne4_oQU240.A_WCYCL1850.anvil_intel
* PET_Ln9_PS.ne30_oECv3_ICG.A_WCYCL1850S.anvil_intel.allactive-mach-pet

**Comment by jonbob** [comment]
merged to next

**Comment by jonbob** [comment]
merged to master and expected NML DIFFs blessed

**Pull request merged**